### PR TITLE
gnome-inform7: init at unstable-2021-04-06

### PR DESCRIPTION
**Pull request by fitzgibbon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  nativeBuildInputs =
+    [ meson ninja pkg-config inform7 python3 desktop-file-utils wrapGAppsHook ];
+  buildInputs = [
+    gettext
+    libossp_uuid
+    gnome3.gtk
+    gnome3.gtksourceview
+    gnome3.gspell
+    webkit
+    gtkspell3
```
Comment (jtojnar) [inline pkgs/applications/editors/gnome-inform7/default.nix:71]: Does it really use gspell and gtkspell at the same time?

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    patchShebangs src/generate-resource-xml.py
+    patchShebangs src/generate-resource-xml.sh
+  '';
+
+  meta = with lib; {
+    description = "Inform 7 for the Gnome platform";
+    longDescription =
+      "This version of Inform 7 for the Gnome platform was created by Philip Chimento, based on a design by Graham Nelson and Andrew Hunter.";
+    homepage = "https://github.com/ptomato/gnome-inform7";
+    license = licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/editors/gnome-inform7/default.nix:90]: `gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    '';
+  });
+
+in stdenv.mkDerivation {
+  name = "gnome-inform7";
+  version = "unstable-2021-04-06";
+  src = fetchFromGitHub {
+    owner = "ptomato";
+    repo = "gnome-inform7";
+    rev = "c37e045c159692aae2e4e79b917e5f96cfefa66a";
```
Comment (jtojnar) [inline pkgs/applications/editors/gnome-inform7/default.nix:59]: Please add a comment that this is gtk3 branch.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -11479,4 +11479,10 @@
     github = "zupo";
     githubId = 311580;
   };
+  nfitzgibbon = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11482]: Please put this into a separate commit with the message ``maintainers: add nfitzgibbon`` and sort it alphabetically.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      gnome3.gtk
+      gobjectIntrospection
+      gnome3.vala
+      gnome2.gtk-doc
+      wrapGAppsHook
+    ];
+  });
+
+  chimara = (stdenv.mkDerivation {
+    name = "chimara";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:28]: pname + version should be used.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    src = fetchFromGitHub {
+      owner = "chimara";
+      repo = "Chimara";
+      rev = "9934b142af508c75c0f1eed597990f39495b1af4";
+      sha256 = "aRz1XX8XaSLTBIrMIIMS3QNMm6Msi+slrZ6+KYlyRMo=";
+    };
+    nativeBuildInputs =
+      [ meson ninja pkg-config perl bison texinfo python3 glib wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:36]:
```suggestion
    nativeBuildInputs = [ meson ninja pkg-config perl bison texinfo python3 glib wrapGAppsHook ];
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+      gst_all_1.gst-plugins-bad
+      glib
+    ];
+    preConfigure = ''
+      patchShebangs build-aux/meson_post_install.py
+    '';
+  });
+
+in stdenv.mkDerivation {
+  name = "gnome-inform7";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:54]:
```suggestion
  pname = "gnome-inform7";
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    cp ${inform7}/libexec/ni ./src/ni
+    patchShebangs build-aux/*
+    patchShebangs src/generate-resource-xml.py
+    patchShebangs src/generate-resource-xml.sh
+  '';
+
+  meta = with lib; {
+    description = "Inform 7 for the Gnome platform";
+    longDescription =
+      "This version of Inform 7 for the Gnome platform was created by Philip Chimento, based on a design by Graham Nelson and Andrew Hunter.";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:88]:
```suggestion
    longDescription = ''
      This version of Inform 7 for the Gnome platform was created by Philip Chimento, based on a design by Graham Nelson and Andrew Hunter.
    '';
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+      repo = "ratify";
+      rev = "f4d2d60ec73d5588e953650b3879e69a727f30ca";
+      sha256 = "eRh/9pYvdfbdbdJQ7pYMLq5p91I+rtyb/AqEGfakjKs=";
+    };
+    nativeBuildInputs = [ meson ninja pkg-config docbook2x docbook_xml_xslt ];
+    buildInputs = [
+      gnome3.gtk
+      gobjectIntrospection
+      gnome3.vala
+      gnome2.gtk-doc
```
[the remark this answers is not in the document]
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Fixed all these obsolete package names. This is my first contribution so I'm learning as I go, sorry -- is there a convenient reference somewhere to work out which names are deprecated and which are preferred?

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+let
+  ratify = (stdenv.mkDerivation {
+    name = "ratify";
+    src = fetchFromGitHub {
+      owner = "ptomato";
+      repo = "ratify";
+      rev = "f4d2d60ec73d5588e953650b3879e69a727f30ca";
+      sha256 = "eRh/9pYvdfbdbdJQ7pYMLq5p91I+rtyb/AqEGfakjKs=";
+    };
+    nativeBuildInputs = [ meson ninja pkg-config docbook2x docbook_xml_xslt ];
```
[the remark this answers is not in the document]
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Changes these package names.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+  };
+  nativeBuildInputs =
+    [ meson ninja pkg-config inform7 python3 desktop-file-utils wrapGAppsHook ];
+  buildInputs = [
+    gettext
+    libossp_uuid
+    gnome3.gtk
+    gnome3.gtksourceview
+    gnome3.gspell
+    webkit
```
[the remark this answers is not in the document]
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Changes these package names too.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+  name = "gnome-inform7";
+  version = "unstable-2021-04-06";
+  src = fetchFromGitHub {
+    owner = "ptomato";
+    repo = "gnome-inform7";
+    rev = "c37e045c159692aae2e4e79b917e5f96cfefa66a";
+    sha256 = "Q4xoITs3AYXhvpWaABRAvJaUWTtUl8lYQ1k9zX7FrNw=";
+  };
+  nativeBuildInputs =
+    [ meson ninja pkg-config inform7 python3 desktop-file-utils wrapGAppsHook ];
```
[the remark this answers is not in the document]
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Reformatted as suggested.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 1 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: No, it only required gspell. I had left in gtkspell by mistake, sorry! Removed.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 2 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: I couldn't find any reference to newer versions in the software itself so have flagged this as gpl3Only.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 3 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix:78]: Done.

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (fitzgibbon) [reply maintainers/maintainer-list.nix]: This was in a separate commit within this PR -- I've now also changed the handle to match my github username and alphabetized it in the latest commit.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 5 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Changed both of these to pname + version.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 6 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: I have reformatted this entire list to one item per line as suggested by jtojnar above.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 7 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Changed as suggested.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 8 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Changed to multi-line string as suggested.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/applications/editors/gnome-inform7/default.nix]: You should not use packages listed in [aliases.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix
) or bellow [this line](https://github.com/NixOS/nixpkgs/blob/78f2bae18d557bba5f12ab95c0b4001314d6fddd/pkgs/desktops/gnome/default.nix#L281) for <code>gnome~~3~~</code>.

CI will be red when you use them. You can also check for that by having `{allowAliases = false;}` in `~/.config/nixpkgs/config.nix`.

[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Generally C libraries are in top-level namespace these days so you would try `all-packages.nix` first. I would also recommend package search but it looks like it actually prefers the deprecated aliases. Yikes, opened https://github.com/NixOS/nixos-search/issues/306.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+  nativeBuildInputs =
+    [
+      meson
+      ninja
+      pkg-config
+      inform7
+      python3
+      desktop-file-utils
+      wrapGAppsHook
+    ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:90]:
```suggestion
  nativeBuildInputs = [
    meson
    ninja
    pkg-config
    inform7
    python3
    desktop-file-utils
    wrapGAppsHook
  ];
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+        meson
+        ninja
+        pkg-config
+        perl
+        bison
+        texinfo
+        python3
+        glib
+        wrapGAppsHook
+      ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:54]:
```suggestion
    nativeBuildInputs = [
      meson
      ninja
      pkg-config
      perl
      bison
      texinfo
      python3
      glib
      wrapGAppsHook
    ];
```

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 13 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Amended commit with these format changes. Thanks!

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 14 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Amended commit with these format changes. Thanks!

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Thanks -- I had been use the nix package search which was giving me these deprecated packages.

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+    patchShebangs build-aux/*
+    patchShebangs src/generate-resource-xml.py
+    patchShebangs src/generate-resource-xml.sh
+  '';
+
+  meta = with lib; {
+    description = "Inform 7 for the Gnome platform";
+    longDescription = ''
+      This version of Inform 7 for the Gnome platform was created by Philip Chimento, based on a design by Graham Nelson and Andrew Hunter.
+      '';
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:113]:
```suggestion
    '';
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+    goocanvas2
+    libplist
+    ratify
+    chimara
+  ];
+  preConfigure = ''
+    cp ${inform7}/libexec/ni ./src/ni
+    patchShebangs build-aux/*
+    patchShebangs src/generate-resource-xml.py
+    patchShebangs src/generate-resource-xml.sh
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:106]:
```suggestion
    patchShebangs build-aux/* src/generate-resource-xml.{py,sh}
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+    sha256 = "Q4xoITs3AYXhvpWaABRAvJaUWTtUl8lYQ1k9zX7FrNw=";
+  };
+  nativeBuildInputs = [
+      meson
+      ninja
+      pkg-config
+      inform7
+      python3
+      desktop-file-utils
+      wrapGAppsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:87]:
```suggestion
    meson
    ninja
    pkg-config
    inform7
    python3
    desktop-file-utils
    wrapGAppsHook
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+    nativeBuildInputs = [
+        meson
+        ninja
+        pkg-config
+        perl
+        bison
+        texinfo
+        python3
+        glib
+        wrapGAppsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:52]:
```suggestion
      meson
      ninja
      pkg-config
      perl
      bison
      texinfo
      python3
      glib
      wrapGAppsHook
```

File: pkgs/applications/editors/gnome-inform7/default.nix [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+    ];
+    buildInputs = [
+      gtk3
+      gobject-introspection
+      vala gtk-doc
+      wrapGAppsHook
+    ];
+  });
+
+  chimara = (stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/gnome-inform7/default.nix:34]: Why are those not packaged so that they are available to other packages?

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 15 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Fixed -- is there a tool you're using to format the Nix source here? I used nixfmt but it seems to disagree on the whitespace.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 16 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix]: Fixed.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 15 above)
Comment (jtojnar) [reply pkgs/applications/editors/gnome-inform7/default.nix]: nixpkgs-fmt is what most people prefer. But that is not perfect either.

File: pkgs/applications/editors/gnome-inform7/default.nix (hunk 19 above)
Comment (fitzgibbon) [reply pkgs/applications/editors/gnome-inform7/default.nix:34]: I'm not aware of any other projects that depend on them and they either have no tagged releases or the tagged releases can't be built or used to build gnome-inform7.
